src/reducers/profile-view.js

```diff
@@ -29,7 +29,7 @@ import type {
 
 const profile: Reducer<Profile | null> = (state = null, action) => {
   switch (action.type) {
-    case 'VIEW_PROFILE':
+    case 'PROFILE_LOADED':
```

Comment:
Do we actually need to store the profile in the store here? Could it be loaded via thunk action, and only added to the store via `VIEW_PROFILE`? I think this may make the action and state changes more atomic, and reduce the multiple dispatches. I may be missing some necessary step here, but I think this may be possible.

Reply:
I'm not sure if we really want to do that. Since `VIEW_PROFILE` is now more like a setter for the profile view, and `PROFILE_LOADED` is more like a setter of the profile. We don't set the profile to store in `VIEW_PROFILE` anymore. We can return the profile in `loadProfile` thunk action and then postpone the profile storing until `VIEW_PROFILE` but then we'll have to also return everything else we store in `PROFILE_LOADED` too(pathInZipFile, implementationFilter, transformStacks). I think storing them at that moment is more convenient. What do you think?